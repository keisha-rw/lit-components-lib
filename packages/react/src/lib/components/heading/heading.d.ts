import * as React from 'react';
export declare const PdsHeading: React.FunctionComponent<Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
