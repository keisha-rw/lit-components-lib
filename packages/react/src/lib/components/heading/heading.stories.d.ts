import * as React from 'react';
declare const _default: {
    title: string;
    component: React.FunctionComponent<Partial<{
        variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
        inverted: boolean | undefined;
    }> & {
        ref?: React.Ref<any> | undefined;
    } & {
        style?: React.CSSProperties | undefined;
    } & Required<{
        headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
    }> & Partial<{}> & {
        id?: string | undefined;
        slot?: string | undefined;
        className?: string | undefined;
        children?: React.ReactNode;
    }>;
    tags: string[];
};
export default _default;
export declare const Heading1VariantDisplayDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantDisplayDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantDisplaySmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantHeadlineSmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantHeadlineDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantHeadlineLarge: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantTitleExtraSmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantTitleSmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantTitleDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantTitleLarge: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantLabelSmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantLabelDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantLabelLarge: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantMetaSmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading2VariantMetaDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading3VariantDisplayDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading4VariantDisplaySmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading4VariantDisplayDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading5VariantDisplayDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export declare const Heading6VariantDisplayDefault: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Partial<{
    variant: "label-default" | "display-default" | "display-sm" | "headline-lg" | "headline-default" | "headline-sm" | "title-lg" | "title-default" | "title-sm" | "title-xs" | "label-lg" | "label-sm" | "meta-sm" | "meta-default";
    inverted: boolean | undefined;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Required<{
    headingTag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}> & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
