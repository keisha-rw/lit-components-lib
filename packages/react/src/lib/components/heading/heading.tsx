import * as React from 'react';
import { PdsHeading as PdsHeadingElement } from '@keisha/design-system';
import { PdsHeadingProps } from '../../built-component-props';
import { createPdsReactComponent } from '../../create-pds-react-component';

export const PdsHeading = createPdsReactComponent<PdsHeadingProps>()({
  tagName: 'pds-heading',
  elementClass: PdsHeadingElement,
  react: React,
});
