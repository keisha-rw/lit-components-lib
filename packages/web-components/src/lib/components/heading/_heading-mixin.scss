/*------------------------------------*\
 #HEADING_MIXIN
\*------------------------------------*/

@mixin heading {
  .pds-c-heading {
    margin: 0;
  }

  .pds-c-heading--display-default {
    @include pds-semantic-display-default;
  }

  .pds-c-heading--display-sm {
    @include pds-semantic-display-sm;
  }

  .pds-c-heading--headline-lg {
    @include pds-semantic-headline-lg;
  }

  .pds-c-heading--headline-default {
    @include pds-semantic-headline-default;
  }

  .pds-c-heading--headline-sm {
    @include pds-semantic-headline-sm;
  }

  .pds-c-heading--title-lg {
    @include pds-semantic-title-lg;
  }

  .pds-c-heading--title-default {
    @include pds-semantic-title-default;
  }

  .pds-c-heading--title-sm {
    @include pds-semantic-title-sm;
  }

  .pds-c-heading--title-xs {
    @include pds-semantic-title-xs;
  }

  .pds-c-heading--label-lg {
    @include pds-semantic-label-lg;
  }

  .pds-c-heading--label-default {
    @include pds-semantic-label-default;
  }

  .pds-c-heading--label-sm {
    @include pds-semantic-label-sm;
  }

  .pds-c-heading--meta-default {
    @include pds-semantic-meta-default;
  }

  .pds-c-heading--meta-sm {
    @include pds-semantic-meta-sm;
  }

  .pds-c-heading--inverted {
    color: var(--pds-semantic-text-inverted-default);
  }
}
