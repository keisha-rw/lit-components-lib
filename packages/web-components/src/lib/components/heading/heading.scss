@import './heading-mixin';

/*------------------------------------*\
 #HEADING
\*------------------------------------*/

:host {
  display: block;
}

@include heading;
