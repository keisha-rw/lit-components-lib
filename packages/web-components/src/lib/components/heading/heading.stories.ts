import { Meta, StoryObj } from '@storybook/web-components';
import { html, nothing } from 'lit';
import './heading';

export default {
  title: 'Components/Heading',
  component: 'pds-heading',
  tags: ['autodocs'],
  argTypes: {
    variant: {
      control: 'select',
      options: [
        'display-default',
        'display-sm',
        'headline-lg',
        'headline-default',
        'headline-sm',
        'title-lg',
        'title-default',
        'title-sm',
        'title-xs',
        'label-lg',
        'label-default',
        'label-sm',
        'meta-sm',
        'meta-default',
      ],
    },
    headingTag: {
      control: 'select',
      options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'],
    },
    inverted: {
      control: 'boolean',
    },
  },
  render: (args) =>
    html`<pds-heading
      headingTag="${args.headingTag || nothing}"
      variant="${args.variant || nothing}"
      >Some heading text goes here</pds-heading
    >`,
} as Meta;

// span element is needed to wrap plain text slots https://github.com/storybookjs/storybook/issues/15963

export const Heading1VariantDisplayDefault: StoryObj = {
  name: 'H1 variant, display default',
  args: { headingTag: 'h1', variant: 'display-default' },
};

export const Heading2VariantDisplayDefault: StoryObj = {
  name: 'H2 variant, display default',
  args: { headingTag: 'h2', variant: 'display-default' },
};

export const Heading2VariantDisplaySmall: StoryObj = {
  name: 'H2 variant, display small',
  args: { headingTag: 'h2', variant: 'display-sm' },
};

export const Heading2VariantHeadlineSmall: StoryObj = {
  name: 'H2 variant, headline small',
  args: { headingTag: 'h2', variant: 'headline-sm' },
};

export const Heading2VariantHeadlineDefault: StoryObj = {
  name: 'H2 variant, headline default',
  args: { headingTag: 'h2', variant: 'headline-default' },
};

export const Heading2VariantHeadlineLarge: StoryObj = {
  name: 'H2 variant, headline large',
  args: { headingTag: 'h2', variant: 'headline-lg' },
};

export const Heading2VariantTitleExtraSmall: StoryObj = {
  name: 'H2 variant, title xsmall',
  args: { headingTag: 'h2', variant: 'title-xs' },
};

export const Heading2VariantTitleSmall: StoryObj = {
  name: 'H2 variant, title small',
  args: { headingTag: 'h2', variant: 'title-sm' },
};

export const Heading2VariantTitleDefault: StoryObj = {
  name: 'H2 variant, title default',
  args: { headingTag: 'h2', variant: 'title-default' },
};

export const Heading2VariantTitleLarge: StoryObj = {
  name: 'H2 variant, title large',
  args: { headingTag: 'h2', variant: 'title-lg' },
};

export const Heading2VariantLabelSmall: StoryObj = {
  name: 'H2 variant, label small',
  args: { headingTag: 'h2', variant: 'label-sm' },
};

export const Heading2VariantLabelDefault: StoryObj = {
  name: 'H2 variant, label default',
  args: { headingTag: 'h2', variant: 'label-default' },
};

export const Heading2VariantLabelLarge: StoryObj = {
  name: 'H2 variant, label large',
  args: { headingTag: 'h2', variant: 'label-lg' },
};

export const Heading2VariantMetaSmall: StoryObj = {
  name: 'H2 variant, meta small',
  args: { headingTag: 'h2', variant: 'meta-sm' },
};

export const Heading2VariantMetaDefault: StoryObj = {
  name: 'H2 variant, meta default',
  args: { headingTag: 'h2', variant: 'meta-default' },
};

export const Heading3VariantDisplayDefault: StoryObj = {
  name: 'H3 variant, display default',
  args: { headingTag: 'h3', variant: 'display-default' },
};

export const Heading4VariantDisplaySmall: StoryObj = {
  name: 'H4 variant, display small',
  args: { headingTag: 'h4', variant: 'display-sm' },
};

export const Heading4VariantDisplayDefault: StoryObj = {
  name: 'H4 variant, display default',
  args: { headingTag: 'h4', variant: 'display-default' },
};

export const Heading5VariantDisplayDefault: StoryObj = {
  name: 'H5 variant, display default',
  args: { headingTag: 'h5', variant: 'display-default' },
};

export const Heading6VariantDisplayDefault: StoryObj = {
  name: 'H6 variant, display default',
  args: { headingTag: 'h6', variant: 'display-default' },
};
