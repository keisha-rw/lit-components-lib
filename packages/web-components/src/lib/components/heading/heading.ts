import { html } from 'lit';
import { property } from 'lit/decorators.js';
import { pdsCustomElement as customElement } from '../../decorators/pds-custom-element';
import { PdsElement } from '../PdsElement';
import { required } from '../../decorators/required';
import styles from './heading.scss?inline';

/**
 * @summary This component displays different heading tags in different sizes based on properties
 *
 * @slot default The heading text
 */
@customElement('pds-heading', {
  category: 'component',
  type: 'component',
  state: 'stable',
  styles,
})
export class PdsHeading extends PdsElement {
  /**
   * Style variant
   * - **display-default** renders a heading with `display-default` type semantic preset
   * - **display-sm** renders a heading with `display-sm` type semantic preset
   * - **headline-lg** renders a heading with `headline-lg` type semantic preset
   * - **headline-default** renders a heading with `headline-default` type semantic preset
   * - **headline-sm** renders a heading with `headline-sm` type semantic preset
   * - **title-lg** renders a heading with `title-lg` type semantic preset
   * - **title-default** renders a heading with `title-default` type semantic preset
   * - **title-sm** renders a heading with `title-sm` type semantic preset
   * - **title-xs** renders a heading with `title-xs` type semantic preset
   * - **label-lg** renders a heading with `label-lg` type semantic preset
   * - **label-default** renders a heading with `label-default` type semantic preset
   * - **label-sm** renders a heading with `label-sm` type semantic preset
   * - **meta-default** renders a heading with `meta-default` type semantic preset
   * - **meta-sm** renders a heading with `meta-sm` type semantic preset
   */
  @property()
  variant:
    | 'display-default'
    | 'display-sm'
    | 'headline-lg'
    | 'headline-default'
    | 'headline-sm'
    | 'title-lg'
    | 'title-default'
    | 'title-sm'
    | 'title-xs'
    | 'label-lg'
    | 'label-default'
    | 'label-sm'
    | 'meta-sm'
    | 'meta-default';

  @property({ type: Boolean })
  inverted?: boolean;

  /**
   * Dynamic tag name for the component
   * This is needed to use proper semantic heading treatments depending on where the banner lives on the page
   * <pds-heading variant="display-default">
   *   <ul>
   *     <li>**h1** renders an `h1` tag</li>
   *     <li>**h2** renders an `h2` tag</li>
   *     <li>**h3** renders an `h3` tag</li>
   *     <li>**h4** renders an `h4` tag</li>
   *     <li>**h5** renders an `h5` tag</li>
   *     <li>**h6** renders an `h6` tag</li>
   *   </ul>
   * </pds-heading>
   */
  @required
  @property()
  headingTag: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

  /**
   * @internal
   */
  get classNames() {
    return {
      [this.variant]: !!this.variant,
      inverted: this.inverted === true,
    };
  }

  render() {
    switch (this.headingTag) {
      case 'h2':
        return html`<h2 class=${this.getClass()}>
          <slot></slot>
        </h2> `;
      case 'h3':
        return html`<h3 class=${this.getClass()} part="heading">
          <slot></slot>
        </h3>`;
      case 'h4':
        return html`<h4 class=${this.getClass()}>
          <slot></slot>
        </h4>`;
      case 'h5':
        return html`<h5 class=${this.getClass()}>
          <slot></slot>
        </h5>`;
      case 'h6':
        return html`<h6 class=${this.getClass()}>
          <slot></slot>
        </h6>`;
      default:
        return html`<h1 class=${this.getClass()}>
          <slot></slot>
        </h1>`;
    }
  }
}
