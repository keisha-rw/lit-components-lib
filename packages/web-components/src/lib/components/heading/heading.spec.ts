import * as path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import { assert, fixture } from '@open-wc/testing';
import { PdsHeading } from './heading';
import createSnapshots from '../../../../test/utils/snapshots';

initStoryshots({
  storyKindRegex: /Heading/,
  framework: 'web-components',
  configPath: path.join(__dirname, '../../../../.storybook'),
  asyncJest: true,
  test: createSnapshots,
});

describe('Heading unit tests', () => {
  let element: Element;

  beforeEach(async () => {
    element = await fixture(
      '<pds-heading headingTag="h1" variant="display-default">Heading</pds-heading>',
    );
  });

  it('is an instance of PdsHeading', () => {
    expect(element).toBeInstanceOf(PdsHeading);
  });

  it('renders a h1 element when the headingTag = h1', async () => {
    element = (await fixture(
      '<pds-heading headingTag="h1" variant="display-default">Heading 1, display-default variant</pds-heading>',
    )) as PdsHeading;

    expect(element).toMatchSnapshot();
  });

  it('renders a h2 element when the headingTag = h2', async () => {
    element =
      await fixture(`<pds-heading headingTag="h2" variant="display-default"
    >Heading 2, display-default variant</pds-heading>`);

    expect(element).toMatchSnapshot();
  });

  it('renders a h3 element when the headingTag = h3', async () => {
    element =
      await fixture(`<pds-heading headingTag="h3" variant="display-default"
    >Heading 3, display-default variant</pds-heading>`);

    expect(element).toMatchSnapshot();
  });

  it('renders a h4 element when the headingTag = h4', async () => {
    element =
      await fixture(`<pds-heading headingTag="h4" variant="display-default"
    >Heading 4, display-default variant</pds-heading>`);

    expect(element).toMatchSnapshot();
  });

  it('renders a h5 element when the headingTag = h5', async () => {
    element =
      await fixture(`<pds-heading headingTag="h5" variant="display-default"
    >Heading 5, display-default variant</pds-heading>`);

    expect(element).toMatchSnapshot();
  });

  it('renders a h6 element when the headingTag = h6', async () => {
    element =
      await fixture(`<pds-heading headingTag="h6" variant="display-default"
    >Heading 6, display-default variant</pds-heading>`);

    expect(element).toMatchSnapshot();
  });

  it('is accessible', async () => {
    await assert.isAccessible(element);
  });
});
