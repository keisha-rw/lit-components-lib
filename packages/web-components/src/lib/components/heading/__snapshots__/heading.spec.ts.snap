// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading unit tests renders a h1 element when the headingTag = h1 1`] = `
<pds-heading
  headingtag="h1"
  variant="display-default"
>
  Heading 1, display-default variant
</pds-heading>
`;

exports[`Heading unit tests renders a h2 element when the headingTag = h2 1`] = `
<pds-heading
  headingtag="h2"
  variant="display-default"
>
  Heading 2, display-default variant
</pds-heading>
`;

exports[`Heading unit tests renders a h3 element when the headingTag = h3 1`] = `
<pds-heading
  headingtag="h3"
  variant="display-default"
>
  Heading 3, display-default variant
</pds-heading>
`;

exports[`Heading unit tests renders a h4 element when the headingTag = h4 1`] = `
<pds-heading
  headingtag="h4"
  variant="display-default"
>
  Heading 4, display-default variant
</pds-heading>
`;

exports[`Heading unit tests renders a h5 element when the headingTag = h5 1`] = `
<pds-heading
  headingtag="h5"
  variant="display-default"
>
  Heading 5, display-default variant
</pds-heading>
`;

exports[`Heading unit tests renders a h6 element when the headingTag = h6 1`] = `
<pds-heading
  headingtag="h6"
  variant="display-default"
>
  Heading 6, display-default variant
</pds-heading>
`;
